From 81c99e59607ceaaf831a482cf6d27262c380bc5c Mon Sep 17 00:00:00 2001
From: "carlosgc@webkit.org" <carlosgc@webkit.org>
Date: Sun, 3 Dec 2017 09:07:25 +0000
Subject: [PATCH 1/3] GstCaps leaked in
 MediaPlayerPrivateGStreamer::createGSTPlayBin
 https://bugs.webkit.org/show_bug.cgi?id=180311

Reviewed by Michael Catanzaro.

gst_caps_new_empty_simple() is passed directly to g_object_set(). The returned caps should be freed with
gst_caps_unref().

* platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::createGSTPlayBin):

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@225458 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 .../platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp        | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
index 76948c436c4..60fb09dc0b2 100644
--- a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
@@ -2223,7 +2223,8 @@ void MediaPlayerPrivateGStreamer::createGSTPlayBin()
     m_textAppSinkPad = adoptGRef(gst_element_get_static_pad(m_textAppSink.get(), "sink"));
     ASSERT(m_textAppSinkPad);
 
-    g_object_set(m_textAppSink.get(), "emit-signals", true, "enable-last-sample", false, "caps", gst_caps_new_empty_simple("text/vtt"), nullptr);
+    GRefPtr<GstCaps> textCaps = adoptGRef(gst_caps_new_empty_simple("text/vtt"));
+    g_object_set(m_textAppSink.get(), "emit-signals", TRUE, "enable-last-sample", FALSE, "caps", textCaps.get(), nullptr);
     g_signal_connect_swapped(m_textAppSink.get(), "new-sample", G_CALLBACK(newTextSampleCallback), this);
 
     g_object_set(m_pipeline.get(), "text-sink", m_textAppSink.get(), nullptr);
-- 
2.11.0

